Add list of critical success factors for etcd replication (#2021)
sensu/sensu-docs: content/sensu-go/5.16/reference/etcdreplicators.md

{{< /language-toggle >}}

## Critical success factors for etcd replication

> Before you implement etcd replicators, review these must-know details that are critical to your success.

Author: ```suggestion
Before you implement etcd replicators, review these details &mdash; they are critical to your success.
```

**Bind your etcd listener to an external port that is *not* the default.**

- Replication will not work if you bind your etcd listener to the default port.

**Use only addresses that clients can route to for `etcd-client-advertise-urls`.**

- If you use addresses that clients cannot route to for `etcd-client-advertise-urls`, replication may be inconsistent: it may work at first but then stop working later.

**Put the certificate and key of the follower cluster in files that the leader can access.**

- If the leader cannot access the follower cluster files that contain the certificate and key, replication will not work.

**For self-signed certificates, supply the CA certificate in the replicator definition.**

- If you have a self-signed certificate and you do not supply the CA certificate in the replicator definition, replication will not work.

**If you're using insecure mode, use TLS mutual authentication.**

- Never use insecure mode without TLS mutual authentication outside of a testbed.

> _**WARNING**: Make sure to confirm your configuration. The server will accept incorrect EtcdReplicator definitions without sending a warning. If your configuration is wrong, replication will not work._

Author: ```suggestion
_**WARNING**: Make sure to confirm your configuration. The server will accept incorrect EtcdReplicator definitions without sending a warning. If your configuration is incorrect, replication will not work._
```

{{< /language-toggle >}}

## Critical success factors for etcd replication

Before you implement etcd replicators, review these must-know details that are critical to your success.

**Bind your etcd listener to an external port that is *not* the default.**

- Replication will not work if you bind your etcd listener to the default port.

**Use only addresses that clients can route to for `etcd-client-advertise-urls`.**

> - If you use addresses that clients cannot route to for `etcd-client-advertise-urls`, replication my be inconsistent: it may work at first but then stop working later.

Author: ```suggestion
- If you use addresses that clients cannot route to for `etcd-client-advertise-urls`, replication may be inconsistent: it may work at first but then stop working later.
```
